tests/integrations/subprocess/test_Subprocess__stream_output.py

```diff
@@ -63,22 +50,17 @@ def test_keyboard_interrupt(mock_sub, popen_process, capsys):
     """KeyboardInterrupt is suppressed if user sends CTRL+C and all output is
     printed."""
 
-    def slow_poll(*a):
-        sleep(0.5)
-        return -3
-
-    # this helps ensure that the output streaming thread doesn't
-    # finish before is_alive() is called and therefore ensures
-    # that stop_func is always executed during this test.
-    popen_process.poll = mock.MagicMock()
-    popen_process.poll.side_effect = slow_poll
-
     send_ctrl_c = mock.MagicMock()
     send_ctrl_c.side_effect = [False, KeyboardInterrupt]
 
     mock_sub.stream_output("testing", popen_process, stop_func=send_ctrl_c)
 
-    assert capsys.readouterr().out == ("output line 1\n" "\n" "output line 3\n")
+    assert (
+        capsys.readouterr().out == "output line 1\n"
+        "\n"
+        "output line 3\n"
+        "Stopping...\n"
```

Comment:
Ok - this is really weird; I'm not seeing the CI failure on any of my test configurations. Investigating...

Reply:
Found the source of the problem. Since the streamer thread isn't doing any substantial work, it was completely draining and exiting before we did our second stop check. The emergence of this behavior will be completely non-deterministic, as it depends on OS-level thread scheduling; however, we can strongly encourage the right behavior by adding a sleep to gently suggest to the OS that the streamer thread should suspend for a bit.